common/src/main/java/com/wynntils/features/tooltips/ItemCompareFeature.java

```diff
@@ -4,162 +4,648 @@
+                    lines,
+                    x,
+                    y,
+                    tooltipWidth,
+                    tooltipHeight,
+                    tooltipScaleFactor,
+                    pair.value().getTooltipImage()));
+            prevX = x;
+        }
+
+        // Fix tooltips overflow by rendering tooltips in two rows and slightly scaling them down, if possible
+        if (!tooltips.isEmpty() && tooltips.getLast().getX() < 0) {
+            fixTooltipOverflow(tooltips, window.getGuiScaledHeight());
+        }
+
+        // Centering all tooltips horizontally
+        int d1 = tooltips.getLast().getX();
+        int d2 = window.getGuiScaledWidth() - (hoveredX + hoveredTooltipWidth);
+        int offsetX = Math.floorDiv((d1 + d2), 2) - Math.max(d1, d2);
+        tooltips.forEach(tooltip -> tooltip.offsetX(offsetX));
+        hoveredX += offsetX;
+
+        event.setCanceled(true);
+
+        changePositioner = true;
+        poseStack.pushPose();
+        poseStack.scale(hoveredScaleFactor, hoveredScaleFactor, 1);
+        guiGraphics.renderTooltip(
+                font, hoveredLines, hoveredItemStack.getTooltipImage(), (int) (hoveredX / hoveredScaleFactor), (int)
+                        (hoveredY / hoveredScaleFactor));
+        poseStack.popPose();
+
+        for (Tooltip tooltip : tooltips) {
+            poseStack.pushPose();
+            float scaleFactor = tooltip.getScaleFactor();
+            poseStack.scale(scaleFactor, scaleFactor, 1);
             guiGraphics.renderTooltip(
-                    FontRenderer.getInstance().getFont(), itemToCompare, mouseX + hoveredWidth + 10, mouseY);
+                    font,
+                    tooltip.getLines(),
+                    tooltip.getVisualTooltipComponent(),
+                    (int) (tooltip.getX() / scaleFactor),
+                    (int) (tooltip.getY() / scaleFactor));
+            poseStack.popPose();
         }
+        changePositioner = false;
 
         poseStack.popPose();
     }
 
+    private boolean isMatchingType(WynnItem wynnItem, GearTypeItemProperty gearItemReference) {
+        if (wynnItem instanceof GearItem gearItem) {
+            return gearItem.getGearType() == gearItemReference.getGearType();
+        }
+        return false;
+    }
+
     private boolean isMatchingType(ItemStack itemStack, GearTypeItemProperty gearItemReference) {
         Optional<GearTypeItemProperty> gearOpt = Models.Item.asWynnItemProperty(itemStack, GearTypeItemProperty.class);
-        if (gearOpt.isEmpty()) return false;
+        return gearOpt.isEmpty() ? false : gearOpt.get().getGearType() == gearItemReference.getGearType();
+    }
+
+    private void onSelectKeyPress(Slot hoveredSlot) {
+        if (hoveredSlot == null) return;
+
+        int slot = hoveredSlot.getContainerSlot();
+
+        // Selecting armor and accessory slots is not allowed
+        for (int i : InventoryAccessory.getSlots()) {
+            if (slot == i) return;
+        }
```

Comment:
What's the reason for this? If I wanted to compare my necklace with one in a container this forces me to select the container one which for some could be a minor inconvenience

Reply:
If by 'my necklace' you mean the one you have equipped, it is being compared against already. To compare with a necklace in a container just hold compare button over it

Reply:
Since equipped items are added to `itemsToCompare` first and only then selected items are added to it (if not present already) equipped items being selected will never change anything, so I just disabled selecting them. And the reason why I check if selected item is present in `itemsToCompare` is because held item (weapon) can be selected and not allowing to select held item is kinda weird (you can select it and then take it in your hand anyways)